pkg/clustermanager/cluster_manager.go

```diff
@@ -689,6 +694,56 @@ func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster,
 	return nil
 }
 
+// UpgradeClusterAPI incorporates EKS-A cluster status changes for cluster upgrade process.
+func (c *ClusterManager) UpgradeClusterAPI(ctx context.Context, managementCluster, workloadCluster *types.Cluster, newClusterSpec *cluster.Spec, provider providers.Provider) error {
+	logger.V(3).Info("check generation and observed generation numbers")
+
+	isStatusUpdated := func() error {
+		eksaCluster, err := c.clusterClient.GetEksaCluster(ctx, managementCluster, managementCluster.Name)
```

Comment:
I would prefer if we used a kubernetes.Client here :) with the `.Get` method :)

Reply:
GetEksaCluster used Get method to get the cluster details. https://github.com/aws/eks-anywhere/blob/main/pkg/executables/kubectl.go#L1500C25-L1500C25. Let me know if you want this any different way.